crates/nu-cli/src/completions/completer.rs

```diff
@@ -348,8 +348,43 @@ impl NuCompleter {
                 for (arg_idx, arg) in call.arguments.iter().enumerate() {
                     let span = arg.span();
                     if span.contains(pos) {
-                        // if customized completion specified, it has highest priority
-                        if let Some(decl_id) = arg.expr().and_then(|e| e.custom_completion) {
+                        // Get custom completion from PositionalArg or Flag
+                        let custom_completion_decl_id = {
+                            // Check PositionalArg or Flag from Signature
+                            let signature = working_set.get_decl(call.decl_id).signature();
```

Comment:
The only concern I have is with the way our `Command::signature` and `Signature` itself work by creating some stuff again and again on the fly. But if this lookup isn't too costly it feels like the more sensible way to go forward.